src/main/java/ru/geekbrains/VaolEr/dao/ProductDAO.java

```diff
@@ -0,0 +1,93 @@
+package ru.geekbrains.VaolEr.dao;
+
+
+import org.springframework.stereotype.Repository;
+import ru.geekbrains.VaolEr.config.EntityManagerConfig;
+import ru.geekbrains.VaolEr.model.Buyer;
+import ru.geekbrains.VaolEr.model.Product;
+
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.TypedQuery;
+import java.util.List;
+
+@Repository
+public class ProductDAO {
+
+    private final EntityManagerConfig entityManagerConfig;
+    private EntityManager entityManager;
+
+    public ProductDAO(EntityManagerConfig entityManagerConfig) {
+        this.entityManagerConfig = entityManagerConfig;
+        this.entityManager = this.entityManagerConfig.getEntityManager();
+    }
+
+    public boolean create( Product product){
+        try{
+            //Create
+            this.entityManager.getTransaction().begin();
+            this.entityManager.persist(product);
+            this.entityManager.getTransaction().commit();
+            return true;
+        } catch (Exception e){
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+     public Product findById(Long id){
+        //Read
+        //entityManager.getTransaction().begin();
+        return this.entityManager.find(Product.class, id);
+    }
+
+
+    public List<Product> findAll(){
+        this.entityManager.getTransaction().begin();
+        List<Product> list = (List<Product>) this.entityManager.createQuery("SELECT p from Product p")
+                .getResultList();
+        this.entityManager.getTransaction().commit();
+        return list;
+    }
+
+    public Product saveOrUpdate(Product product) {
+        //Update
+        this.entityManager.getTransaction().begin();
+        Product productFromDb = findById(product.getId());
+        productFromDb.setName(product.getName());
+        productFromDb.setCost(product.getCost());
+        this.entityManager.merge(productFromDb);
+        this.entityManager.getTransaction().commit();
+        return findById(product.getId());
+    }
+
+    public boolean delete(Product product){
+        try {
+            this.entityManager.getTransaction().begin();
+            Product productFromDb = findById(product.getId());
+            if(productFromDb.equals(null)){
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.